tests/test_noise.py

```diff
@@ -130,3 +138,201 @@ def pinkish_spectrum(x, y):
     # Check we are achieving an estimate near the target
     logger.debug(f"Estimated Noise Variance {estimated_noise_var}")
     assert np.isclose(estimated_noise_var, target_noise_variance, rtol=0.1)
+
+    # Check sampling yields an estimate near target.
+    sample_n = 16
+    sample_res = 32
+    im_zeros = Image(np.zeros((sample_n, sample_res, sample_res)))
+    im_noise_sample = sim_fixture.noise_adder._forward(im_zeros, range(sample_n))
+    sampled_noise_var = np.var(im_noise_sample.asnumpy())
+
+    logger.debug(f"Sampled Noise Variance {sampled_noise_var}")
+    assert np.isclose(sampled_noise_var, target_noise_variance, rtol=0.1)
+
+
+@pytest.mark.parametrize("res", RESOLUTIONS, ids=lambda param: f"res={param}")
+@pytest.mark.parametrize(
+    "target_noise_variance", VARS, ids=lambda param: f"var={param}"
+)
+@pytest.mark.parametrize("dtype", DTYPES, ids=lambda param: f"dtype={param}")
+def test_psnr(res, target_noise_variance, dtype):
+    vol = np.ones((res,) * 3, dtype=dtype)
+    g = grid_3d(res, normalized=False)
+    mask = g["r"] > res // 2
+    vol[mask] = 0
+
+    sim = Simulation(
+        vols=Volume(vol),
+        n=16,
+        amplitudes=1,
+        offsets=0,
+        dtype=dtype,
+        noise_adder=WhiteNoiseAdder(var=target_noise_variance),
+    )
+
+    psnr = sim.estimate_psnr(units="dB")
+    logger.debug(f"PSNR target={target_noise_variance} L={sim.L} {sim.dtype} {psnr}")
+    assert np.isclose(psnr, -10 * np.log10(target_noise_variance), rtol=0.01)
```

Comment:
Can you provide a quick explanation comment for why PSNR = SNR in this case?

Reply:
Sure! When the noise is zero centered, total number of pixels is modest, and values of the signal are small wrt to the noise, this essentially is max(N)**2 /  np.mean(N**2) for N gaus(0, sqrt(var)) , but then averaged over stack of images. Hand wavy, it is saying for unit test sized problems we expect the noise to be below twosigma most of the time (3 sigma events would be averaged away), but also expect to see two sigma events regularly (like almost every image sample). Something like that, I think. Been a few weeks now 😇 .

Perhaps not testing the right things, but seemed fair as a smoke test.  Regardless, being deleted anyway...